mindsdb/libs/phases/model_analyzer/model_analyzer.py

```diff
@@ -3,33 +3,62 @@
 from mindsdb.libs.phases.base_module import BaseModule
 from mindsdb.libs.data_types.sampler import Sampler
 from mindsdb.libs.ml_models.pytorch.libs import base_model;
+from mindsdb.libs.ml_models.probabilistic_validator import ProbabilisticValidator
 from mindsdb.libs.ml_models.pytorch.libs.torch_helpers import array_to_float_variable
-import importlib
+from mindsdb.libs.helpers.norm_denorm_helpers import denorm
 
+import pandas as pd
 
+#@TODO: Define a way to save self.transaction.persistent_model_metadata.probabilistic_validator
 class ModelAnalyzer(BaseModule):
 
     phase_name = PHASE_MODEL_ANALYZER
 
     def run(self):
```

Comment:
please add some comments for the code explaining whats been done here

Reply:
Explaination should be found within the newly added comments in the probabilistic validator class and model analyzer phase.

Essentially TinyDB doesn't behave well when a dictionary contains binary, and pickle encoding is weird, so we just pickle each probabilistic validator, then decode it with `latin1` encoding (so that it becomes a python `str` object) and then TinyDB can save that do a file.

Later, when we need to use the model, we can simply unpickle the saved string after encoding it using `latin1` (sinc pickle only works on binary data)

Take a look at the 

`pickle` and `unpickle` methods in ProbabilisticValidator, it's not complex at all, just a bit messy and unintuitive (but it's the easiest way I found to serialize an arbitrary object into TinyDB).

Reply:
i understand, if it works its fine :)